### PR TITLE
Avoid requiring non std-lib spec spec_helper in hooks_spec

### PR DESCRIPTION
**Pull request by bcardiff**
I noticed that the spec added in #9090 is requiring the top most spec_helper which is not aimed to be used in the std-lib specs

cc: @straight-shoota

### CODE REVIEW

**Comment by straight-shoota** [comment]
There are other uses as well (at least in kernel_spec).

If it's not intended to be used in stdlb-specs, it should be moved to compiler specs folder.

**Comment by bcardiff** [comment]
hmmm, you are right it is also used somewhere else. Maybe is the `require "prelude"` that triggers some stuff. Either way with this PR is more similar to what is done in `call_stack_spec.cr` which seems to me a preferred approach and does not complain when the spec is run independantly.

**Comment by straight-shoota** [comment]
Yes, it's probably better that way. But I don't think it makes sense to have a `compile_and_run` helper in every spec, let's extract that in a support helper.

**Comment by bcardiff** [comment]
done

File: spec/std/exception/call_stack_spec.cr [hunk 1]
```diff
@@ -1,17 +1,5 @@
 require "../spec_helper"
```
Comment (straight-shoota) [inline spec/std/exception/call_stack_spec.cr:1]: This file needs to require `./spec_helper`. I'm not sure if `../spec_helper` is necessary.

**Comment by straight-shoota** [comment]
What about `kernel_spec`?

File: spec/std/exception/call_stack_spec.cr (hunk 1 above)
Comment (bcardiff) [reply spec/std/exception/call_stack_spec.cr:1]: I don't follow. That file does not exist.

**Comment by bcardiff** [comment]
done done, and a couple more.

File: spec/std/exception/call_stack_spec.cr (hunk 1 above)
Comment (straight-shoota) [reply spec/std/exception/call_stack_spec.cr:1]: Sry, I mixed up the file location.

**Pull request merged**